homeassistant/components/unifi/sensor.py

```diff
@@ -279,6 +284,71 @@ def make_wan_latency_entity_description(
     )
 
 
+@callback
+def async_device_temperatures_value_fn(
+    temperature_name: str, hub: UnifiHub, device: Device
+) -> float:
+    """Retrieve the temperature of the device."""
+    if device.temperatures:
+        temperature = _device_temperature(temperature_name, device.temperatures)
+        return temperature if temperature is not None else 0
+    return 0
```

Comment:
return 0 is not reachable here as #_supported_fn will make sure to not update the value see entity.UnifiEntity.async_signalling_callback

Maybe something like this

```suggestion
    return_value: float = 0
    if device.temperatures:
        temperature = _device_temperature(temperature_name, device.temperatures)
        return_value = temperature if temperature is not None else 0
    return return_value
```